### PR TITLE
Add cache to getDbColumnTypes() function to improve performances

### PR DESCRIPTION
**Pull request by tonyyb**
If `$this->crud->setFromDb()` is called, application will do one database request per column to get type.
If table has 40 columns, the listing of items can take more than 5 seconds.

We can use Cache to improve application performances (do the requests once).

### CODE REVIEW

**Comment by jsiebach** [comment]
This would require people to clear their cache anytime they make an edit to table schema, would it not?

**Comment by tonyyb** [comment]
Hi @jsiebach, of course this feature require people to clear their cache. 
Unfortunately Laraval doesn't provides an event when migrations are launched.
What do you think about a config to able / disable the cache ? By default able this feature in local env and disable in production env ?

File: src/PanelTraits/AutoSet.php [hunk 1]
```diff
@@ -50,7 +52,10 @@ public function getDbColumnTypes()
         $table_columns = $this->model->getConnection()->getSchemaBuilder()->getColumnListing($this->model->getTable());
 
         foreach ($table_columns as $key => $column) {
-            $column_type = $this->model->getConnection()->getSchemaBuilder()->getColumnType($this->model->getTable(), $column);
+            $cacheNameSuffix = $this->model->getConnection()->getDatabaseName().'-'.$this->model->getTable().'-'.$column;
+            $column_type = Cache::rememberForever('backpack_crud::column_type::'.$cacheNameSuffix, function () use ($column) {
```
Comment (lloy0076) [inline src/PanelTraits/AutoSet.php]: Why forever?

**Comment by lloy0076** [comment]
@tonyyb  - Wouldn't you enable the feature in production but possibly disable in development/debug?

**Comment by jsiebach** [comment]
This might be best as a set of artisan commands. `artisan crud:cache` and `artisan crud:cache-clear`.  It could check the cache for the key in the AutoSet trait, if it's there use it if it's not just pull it from the schema.  But running `artisan crud:cache` would remember them forever and `artisan crud:cache-clear` would forget them all.

**Comment by tabacitu** [comment]
Guys, I might be wrong here, but this might be the same issue as ```hasColumn()``` caching (#1118), which has been patched.

That being said, all CRUDs now use AJAX to get their results, so it's even more difficult to see how slow/fast the querying is. The AJAX response is pretty slow imho, so if this is still an issue, we should consider it. But ```forever``` is a long time. I think something like 1h/6h/24h would be better. Something so small that we don't even need to instruct developers to clear the cache if they change their db schema. What do you think?

**Pull request closed without merging**